Sources/Bot/Services/MergeService.swift

```diff
@@ -751,18 +695,57 @@ extension MergeService.State {
     }
 }
 
-// MARK: - Helpers
+// MARK: - Healthcheck
 
-private extension PullRequest {
+extension MergeService {
+    public final class Healthcheck {
 
-    func isLabelled(with label: PullRequest.Label) -> Bool {
-        return labels.contains(label)
-    }
-    func isLabelled(withOneOf possibleLabels: [PullRequest.Label]) -> Bool {
-        return labels.contains(where: possibleLabels.contains)
+        public enum Reason: Error, Equatable {
+            case potentialDeadlock
+        }
+
+        public enum Status: Equatable {
+            case ok
+            case unhealthy(Reason)
+        }
+
+        public let status: Property<Status>
```

Comment:
[nit] This is the dominant member variable for `class Healthcheck` so pls move to top of scope.

Reply:
Fixed in 6345029c29e9bd24a03d4a99c4e9eba56931e4b5